packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/processor.ts

```diff
@@ -0,0 +1,116 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import { Duration, Size } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+
+/**
+ * Configure the data processor.
+ */
+export interface DataProcessorProps {
+  /**
+   * The length of time Kinesis Data Firehose will buffer incoming data before calling the processor.
+   *s
+   * @default Duration.minutes(1)
+   */
+  readonly bufferInterval?: Duration;
+
+  /**
+   * The amount of incoming data Kinesis Data Firehose will buffer before calling the processor.
+   *
+   * @default Size.mebibytes(3)
+   */
+  readonly bufferSize?: Size;
+
+  /**
+   * The number of times Kinesis Data Firehose will retry the processor invocation after a failure due to network timeout or invocation limits.
+   *
+   * @default 3
+   */
+  readonly retries?: number;
+}
+
+/**
+ * The key-value pair that identifies the underlying processor resource.
+ *
+ * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-kinesisfirehose-deliverystream-processorparameter.html
+ */
+export interface DataProcessorIdentifier {
+  /**
+   * The parameter name that corresponds to the processor resource's identifier.
+   *
+   * Must be an accepted value in `CfnDeliveryStream.ProcessoryParameterProperty.ParameterName`.
+   */
+  readonly parameterName: string;
+
+  /**
+   * The identifier of the underlying processor resource.
+   */
+  readonly parameterValue: string;
+}
+
+/**
+ * The full configuration of a data processor.
+ */
+export interface DataProcessorConfig extends DataProcessorProps {
```

Comment:
I have to say, I don't understand this.

`DataProcessorProps` are creation properties. How can `DataProcessorConfig` extend them...?

Do you perhaps need a separate property on the `IDataProcessor` interface, that returns `DataProcessorProps`?